Update README.md

YanaXu opened the pull request:
The original `README.md` is not well organized. 
This PR is to update it:

- Add a section to list all parameters
- Group all workflow samples into one section
- Add steps to prepare service principal and managed identities before using them for login
- Add examples for login with managed identities

This PR will not be merged till the next release because the workflow of login with managed identities will only work in the new release.

YanaXu commented:
Hi @MS-jeremy , @MoChilia , @jiasli , @isra-fel , @msJinLei , please help to review the update. Thanks.

MS-jeremy commented:
> Hi @MS-jeremy , @MoChilia , @jiasli , @isra-fel , @msJinLei , please help to review the update. Thanks.

@YanaXu I'm not quite sure why I'm not the reviewer of this article, can you check if you @ the wrong person?

YanaXu commented:
> > Hi @MS-jeremy
> >             
> >             Jeremy Li
> >             FTE , @MoChilia
> >             
> >             Shiying Chen
> >             FTE , @jiasli
> >             
> >             Jiashuo Li
> >             FTE , @isra-fel
> >             
> >             Yeming Liu
> >             FTE , @msJinLei
> >             
> >             Lei Jin
> >             FTE , please help to review the update. Thanks.
> 
> @YanaXu Yan Xu FTE I'm not quite sure why I'm the reviewer of this article, can you check if you @ the wrong person?

@MS-jeremy, sorry for this. I mistook you for the other Jeremy Li.

MoChilia commented on README.md:
```diff
@@ -102,154 +157,331 @@ jobs:
+> [!NOTE]
+>
+> "Login With System-assigned Managed Identity" is only supported on GitHub self-hosted runners and the self-hosted runners need to be hosted by Azure virtual machines.
+
```
```suggestion
> [!WARNING]
> Avoid using managed identity login on self-hosted runners in public repositories. Managed identities enable secure authentication with Azure resources and obtain Azure AD tokens without the need for explicit credential management. Any user can open pull requests against your repository and access your self-hosted runners without credentials. See more details in [self-hosted runner security](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security).
```
@jiasli, @Jacekey23, shall we add a warning message for using manage identity on self-hosted runners in public repositories?

MoChilia replied:
cc @dcaro @yonzhan

dcaro replied:
We should have this message at the beginning of this article.

YanaXu replied:
@dcaro, @MoChilia  Is the warning message content OK now?

MoChilia replied:
It's good for me. Agree to have this message at the beginning.

MoChilia commented on README.md:
````diff
@@ -284,11 +535,29 @@ This action doesn't implement ```az logout``` by default at the end of execution
+      Clear-AzContext -Scope Process
+      Clear-AzContext -Scope CurrentUser
 ```
 
````
```suggestion

## Azure CLI output
To suppress the output of Azure CLI commands after logging in, you can set the environment variable `AZURE_CORE_OUTPUT` to `none` at the Workflow, Job or Step level.

```yaml
- name: Azure CLI script
  uses: azure/CLI@v1
  env:
    AZURE_CORE_OUTPUT: none
  with:
    azcliversion: latest
    inlineScript: |
      az account show
      
```

@jiasli, @Jacekey23, shall we add `AZURE_CORE_OUTPUT: none` in each example or add a new section to guide users?

MoChilia replied:
cc @dcaro @yonzhan

isra-fel commented on README.md:
````diff
@@ -1,87 +1,142 @@
 ```
 
-## Sample workflow that uses Azure login action using OIDC to run az cli (Linux)
+It's used in login with a Azure service principal secret.
````
```suggestion
It's used in login with an Azure service principal.
```

isra-fel commented on README.md:
```diff
@@ -1,87 +1,142 @@
 # GitHub Actions for deploying to Azure
 
-## Automate your GitHub workflows using Azure Actions
-
 [GitHub Actions](https://help.github.com/articles/about-github-actions) gives you the flexibility to build an automated software development lifecycle workflow.
```
This doc should focus on the login action. Background information and other actions are less important.

YanaXu replied:
Updated. Update the title and move GitHub Action and other actions' description to the reference section.

isra-fel reviewed (changes requested):
Very rich information here. A few suggestions:

1. A table of content (toc) would be useful to navigate through such a long doc
2. We should be clear about our recommended way of using the action - together with azure cli / ps action. Put it at the beginning.
3. Would be clearer to organize the parameters in a table

YanaXu commented:
> 1. ble of content (toc) would be useful to navigate through such a long

Updated.

isra-fel commented on README.md:
```diff
@@ -1,87 +1,178 @@
+    - [Azure PowerShell Action](#azure-powershell-action)
+  - [Contributing](#contributing)
+
+With the [Azure Login Action](https://github.com/Azure/login), you can login to Azure and run Azure CLI and Azure PowerShell scripts.
```
```suggestion
With the [Azure Login Action](https://github.com/Azure/login), you can login to Azure and run [Azure CLI](link to cli action) and [Azure PowerShell](link to ps action) scripts.
```

YanaXu replied:
Updated.

YanaXu commented:
The Comments are submerged. Here are the links. Other comments are resolved.

- [Yeming's comment](https://github.com/Azure/login/pull/363#pullrequestreview-1716900137)
  - Updated.
- Shiying's comments
  - [about MI warning ](https://github.com/Azure/login/pull/363#discussion_r1384295070) 
    - Added.
  - [about output warning](https://github.com/Azure/login/pull/363#discussion_r1384333383)
